src/pages/AgentNew.tsx

```diff
@@ -25,6 +46,23 @@ export default function AgentNew() {
     setOfficeSearchText('')
   }
 
+  const onSubmit: SubmitHandler<IAddAnAgentForm> = async ({
+    office,
+    ...params
+  }: IAddAnAgentForm) => {
+      const newAgent = await ApiAgent.create({
+        ...params,
+        officeId: office.id,
+      })
+
+      if (newAgent) {
+        toast.success('Agent created')
+        navigate(`/agents/${newAgent.agent.id}`)
+      } else {
+          toast.error('Something went wrong, please try again')
+        }
+  }
+
   useEffect(() => {
```

Comment:
I just noticed while doing a clickthrough that this useEffect fires infinitely, because `handleSearchInput` gets changed every time it is called.

The solution is to move `handleSearchInput` into the body of the useEffect and remove it from the dependency array. (And delete it from line 42):
```
  useEffect(() => {
    const handleSearchInput = debounce(getOffices, 500)

    if (officeSearchText !== '') {
      handleSearchInput(officeSearchText)
    }

    return () => handleSearchInput.cancel()
  }, [officeSearchText])
```